packages/Discretization/test/tstInterpolation.cpp

```diff
@@ -0,0 +1,656 @@
+    cell_topologies_view_host( 5 ) = DTK_QUAD_4;
+    Kokkos::deep_copy( cell_topologies_view, cell_topologies_view_host );
+
+    // Create the points, we are looking for
+    unsigned int const query_offset = 3 * ( ( comm_rank + 1 ) % comm_size );
+    unsigned int constexpr n_points = 4;
+    Kokkos::View<double **, DeviceType> points_coord( "points_coord", n_points,
+                                                      2 );
+    auto points_coord_host = Kokkos::create_mirror_view( points_coord );
+    // First point
+    points_coord_host( 0, 0 ) = query_offset + 1.5;
+    points_coord_host( 0, 1 ) = 0.;
+    // Second point
+    points_coord_host( 1, 0 ) = query_offset + 1.;
+    points_coord_host( 1, 1 ) = 1.5;
+    // Third point
+    points_coord_host( 2, 0 ) = query_offset + 1.5;
+    points_coord_host( 2, 1 ) = 1.5;
+    // Fourth point
+    points_coord_host( 3, 0 ) = query_offset + 2.;
+    points_coord_host( 3, 1 ) = 1.;
+    Kokkos::deep_copy( points_coord_host, points_coord );
+
+    return std::make_tuple( cell_topologies_view, cells, coordinates,
+                            points_coord );
+}
+
+TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( Interpolation, one_topo_three_dim,
+                                   DeviceType )
+{
+    Teuchos::RCP<const Teuchos::Comm<int>> comm =
+        Teuchos::DefaultComm<int>::getComm();
+    int const comm_rank = comm->getRank();
+    unsigned int constexpr dim = 3;
+    Kokkos::View<DTK_CellTopology *, DeviceType> cell_topologies_view;
+    Kokkos::View<unsigned int *, DeviceType> cells;
+    Kokkos::View<double **, DeviceType> coordinates;
+    Kokkos::View<double * [3], DeviceType> points_coord;
+    std::tie( cell_topologies_view, cells, coordinates, points_coord ) =
+        buildHexMesh<DeviceType>( comm );
+
+    // We are now done with building the mesh and we can do the interpolation
+    DataTransferKit::Interpolation<DeviceType> interpolation(
+        comm, cell_topologies_view, cells, coordinates, points_coord );
+
+    Kokkos::View<DataTransferKit::Point *, DeviceType> phys_points(
+        "phys_points" );
+    Kokkos::View<DataTransferKit::Point *, DeviceType> reference_points(
+        "reference_points" );
+    Kokkos::View<bool *, DeviceType> point_in_cell( "pt_in_cell" );
+    interpolation.getReferencePoints( phys_points, reference_points,
+                                      point_in_cell );
+
+    // Check the number of points found on each processor
+    if ( comm_rank == 0 )
+    {
+        TEST_EQUALITY( reference_points.extent( 0 ), 16 );
+    }
+    else if ( comm_rank == 1 )
+    {
+        TEST_EQUALITY( reference_points.extent( 0 ), 16 );
+    }
+    else
+    {
+        TEST_EQUALITY( reference_points.extent( 0 ), 0 );
+    }
+
+    // Reference solution
+    typedef std::array<double, dim> PtCoord;
+    std::map<PtCoord, std::set<PtCoord>> ref_sol;
+    // First point
+    PtCoord pt_1 = {{0.5, 0.5, 0.5}};
+    std::vector<PtCoord> ref_frame_1 = {{{0., 0., 0.}}};
```

Comment:
Stupid Clang warnings...

Reply:
I mean that we cannot write `std::vector<PtCoord> ref_frame_1 = {{0., 0., 0.}};`